nautobot/extras/models/computed_fields.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+
+from django import forms
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.serializers.json import DjangoJSONEncoder
+from django.db import models
+
+# from nautobot.extras.models import ComputedField, CustomField
+# from nautobot.extras.models import ComputedField
+from django.urls import reverse
+from jinja2 import TemplateError
+
+from nautobot.core.models import BaseModel
+from nautobot.extras.models import ChangeLoggedModel
+from nautobot.extras.utils import extras_features, FeatureQuery, TemplateRenderer
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+class ComputedFieldManager(models.Manager.from_queryset(RestrictedQuerySet)):
+    use_in_migrations = True
+
+    def get_for_model(self, model):
+        """
+        Return all CustomFields assigned to the given model.
+        """
+        content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
+        return self.get_queryset().filter(content_type=content_type)
+
+
+@extras_features("graphql")
+class ComputedField(BaseModel, ChangeLoggedModel):
+    """
+    A custom link to an external representation of a Nautobot object. The link text and URL fields accept Jinja2 template
+    code to be rendered with an object as context.
+    """
+
+    content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        limit_choices_to=FeatureQuery("computed_fields"),
+    )
+    name = models.SlugField(max_length=100, unique=True, help_text="Internal field name")
+    label = models.CharField(max_length=100, help_text="Name of the field as displayed to users")
+    description = models.CharField(max_length=200, blank=True)
+    template = models.CharField(max_length=500, help_text="Jinja2 template code for field value")
```

Comment:
I think it would be great to have a way to define the type of data returned : str, int, list 

in the future I can see us adding support for more types like Object

Reply:
If we did add a `type` field similar to custom fields, would we want to enforce that the template for a computed field actually returns that data type?

Reply:
I think the template renderer will always return a string so we'll need to convert the result into the proper type after the rendering. would be good to support str and int at first